client/verta/verta/_deployment/endpoint.py

```diff
@@ -175,10 +175,17 @@ def _update_from_build(self, build_id, strategy, wait=False, resources=None, aut
                 print(".", end='')
                 sys.stdout.flush()
                 time.sleep(5)
-            print()
+
             if self.get_status()['status'] == "error":
+                print()
                 raise RuntimeError("endpoint update failed")
 
+            while len(self.get_status()['components']) > 1:
+                print(".", end='')
+                sys.stdout.flush()
+                time.sleep(5)
+            print()
```

Comment:
I have a vague feeling that a second `while` loop might make us susceptible to a race condition or potentially inconsistent state in the backend.

Do you think it would be possible to combine this condition into the previous while loop (line 174)? Not sure if that would also cause any issues.
Might have to be something like
```python
status_dict = self.get_status()  # if you think of a better var name, please do use it haha
while ((status_dict['status'] == "active" and len(status_dict['components']) > 1)
       or status_dict['status'] == "error"):
```

Reply:
I think I know what you meant. Sth like this:
```
            while status_dict['status'] not in ("active", "error") \
                    or (status_dict['status'] == "active" and len(self.get_status()['components']) > 1):
                print(".", end='')
                status_dict = self.get_status()
                sys.stdout.flush()
                time.sleep(5)
```
Fixed. And test still passed. Please double check to see if the logic makes sense.